src/python/CMSSpark/data_aggregation_plots.py

```diff
@@ -0,0 +1,242 @@
+    x = []
+    for primary_bucket in buckets:
+        bucket = buckets[primary_bucket]
+
+        for key in bucket:
+            if key not in x:
+                x.append(key)
+
+    x = sorted(x)
+
+    ys = {}
+    for primary_bucket in buckets:
+        bucket = buckets[primary_bucket]
+
+        y = []
+        for key in x:
+            if key in bucket:
+                y.append(bucket[key])
+            else:
+                y.append(0)
+
+        ys[primary_bucket] = y
+
+    max_ys = {}
+
+    print('Will draw only ' + str(top_results) + ' results')
+
+    for i in range(0, top_results):
+        max = 0
+        max_key = None
+        for y_key in ys:
+            if y_key not in max_ys:
+                sum_of_values = sum(ys[y_key])
+                if sum_of_values > max or max_key == None:
+                    max = sum_of_values
+                    max_key = y_key
+
+        if max_key != None:
+            max_ys[max_key]=ys[max_key]
+
+    for y_key in max_ys:
+        plt.plot(range(len(x)), max_ys[y_key], label=y_key)
+
+    plt.legend()
+    plt.xticks(range(len(x)), [label[5:] for label in x])
+    plt.xlabel('Date')
+    plt.ylabel('Number of access')
+    ax = plt.gca()
+    ax.get_yaxis().get_major_formatter().set_scientific(False)
+    plt.xticks(rotation=45)
+    if filename == None:
+        plt.show()
+    else:
+        figure = plt.gcf()
+        # Set figure size in inches
+        figure.set_size_inches(8, 6)
+        plt.savefig(filename, dpi=100)
+    plt.close()
+
+
+def make_table(bucket, title1, title2, limit_results=None, filename=None):
+    sorted_bucket_keys = sorted(bucket, key=bucket.get, reverse=True)
+    sum = 0
+    for key in sorted_bucket_keys:
+        sum += int(bucket[key])
+
+    if limit_results != None:
+        sorted_bucket_keys = sorted_bucket_keys[0:limit_results]
+
+    sum_of_table = 0
+    csv = '"' + title1 + '","' + title2 + '","Percentage of all"\n'
+    for key in sorted_bucket_keys:
+        sum_of_table += bucket[key]
+
+        csv += '"' + key +'",'
+        csv += '"' + str((bucket[key])) + '",'
+        csv += '"' + "{0:.4f}".format(100.0 * bucket[key] / sum) + '"'
+        csv += '\n'
+
+    csv += '"Sum","' + str(sum_of_table) + '","' + "{0:.4f}".format(100.0 * sum_of_table / sum) + '"'
+    if filename != None:
+        csv_file = open(filename, 'w')
+        csv_file.write(csv)
+        csv_file.close()
+    else:
+        print (csv)
+
+
+# If record's value of 'column' is not in 'valid_values', change it to 'other_value'
+def filter_values(records, column, valid_values, other_value):
+    new_records = list(records)
+
+    for record in records:
+        if record[column] not in valid_values:
+            record[column] = other_value
+
+    return new_records
+
+
+# Include only T0, T1, T2, T3 site tiers. Set others to 'Other'
+rows = filter_values(rows, 'site_tier', ['T0', 'T1', 'T2', 'T2', 'T3'], 'Other')
```

Comment:
This needs to be put into another function too which will make plots. Then create main function which will parse argument parameters and pass them around into main. As an example of organizing python code please see https://github.com/vkuznet/CMSSpark/blob/master/src/python/CMSSpark/aso_stats.py which defines OptionParser class and use it within a main. Then, it would be easy to use your script by anyone.